### PR TITLE
Pack structures

### PR DESCRIPTION
**Pull request by skhomuti**

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 solc = "0.8.24"
 evm_version = "cancun"
 optimizer = true
-optimizer_runs = 200
+optimizer_runs = 1
```
Comment (dgusakov) [inline foundry.toml:5]: 😢

File: src/interfaces/ICSModule.sol [hunk 2]
```diff
@@ -7,23 +7,22 @@ import { IStakingModule } from "./IStakingModule.sol";
...
+    /* 1 */ uint32 totalVettedKeys; // @dev both increased and decreased
+    /* 1 */ uint32 stuckValidatorsCount; // @dev both increased and decreased
+    /* 1 */ uint32 depositableValidatorsCount; // @dev any value
+    /* 1 */ uint32 targetLimit;
+    /* 1 */ uint8 targetLimitMode;
+    /* 2 */ uint32 totalExitedKeys; // @dev only increased
+    /* 2 */ uint32 refundedValidatorsCount; // @dev only increased
+    /* 2 */ uint32 enqueuedCount; // Tracks how many places are occupied by the node operator's keys in the queue.
+    /* 2 */ address managerAddress;
+    /* 3 */ address proposedManagerAddress;
```
Comment (dgusakov) [inline src/interfaces/ICSModule.sol:23]: I believe these two vars should share the same slot

Comment (madlabman) [reply src/interfaces/ICSModule.sol:23]: Two addresses use 40 bytes in total, so it's impossible.

File: src/interfaces/ICSModule.sol [hunk 3]
```diff
@@ -7,23 +7,22 @@ import { IStakingModule } from "./IStakingModule.sol";
...
+    /* 1 */ uint32 totalWithdrawnKeys; // @dev only increased
+    /* 1 */ uint32 totalDepositedKeys; // @dev only increased
+    /* 1 */ uint32 totalVettedKeys; // @dev both increased and decreased
+    /* 1 */ uint32 stuckValidatorsCount; // @dev both increased and decreased
+    /* 1 */ uint32 depositableValidatorsCount; // @dev any value
+    /* 1 */ uint32 targetLimit;
+    /* 1 */ uint8 targetLimitMode;
+    /* 2 */ uint32 totalExitedKeys; // @dev only increased
+    /* 2 */ uint32 refundedValidatorsCount; // @dev only increased
+    /* 2 */ uint32 enqueuedCount; // Tracks how many places are occupied by the node operator's keys in the queue.
```
Comment (madlabman) [inline src/interfaces/ICSModule.sol:21]: If we change it to something lint `uint64` will it better for us in terms bytecode size >> gas used?

File: test/CSModule.t.sol [hunk 4]
```diff
@@ -2071,7 +2071,7 @@ contract CSMClaimRewards is CSMCommon {
     }
 
     function test_claimRewardsStETH_revertWhenNoNodeOperator() public {
-        vm.expectRevert(CSModule.NodeOperatorDoesNotExist.selector);
+        vm.expectRevert();
```
Comment (dgusakov) [inline test/CSModule.t.sol:2074]: Redundant change

File: test/CSModule.t.sol [hunk 5]
```diff
@@ -2114,7 +2114,7 @@ contract CSMClaimRewards is CSMCommon {
     }
 
     function test_claimRewardsWstETH_revertWhenNoNodeOperator() public {
-        vm.expectRevert(CSModule.NodeOperatorDoesNotExist.selector);
+        vm.expectRevert();
```
Comment (dgusakov) [inline test/CSModule.t.sol:2117]: Redundant change

File: test/CSModule.t.sol [hunk 6]
```diff
@@ -2157,7 +2157,7 @@ contract CSMClaimRewards is CSMCommon {
     }
 
     function test_requestRewardsETH_revertWhenNoNodeOperator() public {
-        vm.expectRevert(CSModule.NodeOperatorDoesNotExist.selector);
+        vm.expectRevert();
```
Comment (dgusakov) [inline test/CSModule.t.sol:2160]: Redundant change

Comment (dgusakov) [reply test/CSModule.t.sol:2160]: Also review test below

File: src/CSModule.sol [hunk 7]
```diff
@@ -57,30 +57,32 @@ contract CSModule is
...
     ICSAccounting public accounting;
+
     ICSEarlyAdoption public earlyAdoption;
-    // @dev max number of Node Operators is limited by uint64 due to Batch serialization in 32 bytes
-    // it seems to be enough
-    // TODO: ^^ comment
-    uint256 private _nodeOperatorsCount; // TODO: pack more efficinetly
-    uint256 private _activeNodeOperatorsCount;
+    uint64 private _nodeOperatorsCount;
+    bool public publicRelease;
```
[the remark this answers is not in the document]
Comment (skhomuti) [reply src/CSModule.sol:71]: because of storage I suppose, but it's ok to fix

File: src/interfaces/ICSModule.sol (hunk 3 above)
Comment (madlabman) [reply src/interfaces/ICSModule.sol:21]: I mean all the values, not the last ones.

File: src/CSModule.sol [hunk 8]
```diff
@@ -1454,7 +1405,8 @@ contract CSModule is
         }
         stuckValidatorsCount = no.stuckValidatorsCount;
         refundedValidatorsCount = no.refundedValidatorsCount;
-        stuckPenaltyEndTimestamp = no.stuckPenaltyEndTimestamp;
+        // @dev unused in CSM
+        stuckPenaltyEndTimestamp = 0;
```
Comment (vgorkavenko) [inline src/CSModule.sol:1409]: doesn't make sense to set value to 0 here. it is already zero by default

File: test/CSModule.t.sol [hunk 9]
```diff
@@ -3196,11 +3196,11 @@ contract CsmGetNodeOperatorNonWithdrawnKeys is CSMCommon {
         assertEq(keys, 2);
     }
 
-    function test_getNodeOperatorNonWithdrawnKeys_RevertWhenNoNodeOperator()
```
Comment (vgorkavenko) [inline test/CSModule.t.sol:3199]: it's worth to have the same for other methods where we removed the check

File: src/CSModule.sol (hunk 8 above)
Comment (skhomuti) [reply src/CSModule.sol:1409]: i'd prefer to have it explicitly

File: test/CSModule.t.sol (hunk 9 above)
Comment (skhomuti) [reply test/CSModule.t.sol:3199]: fixed

**Review by vgorkavenko — APPROVED** [review]

**Review by dgusakov — APPROVED** [review]

**Pull request merged**